### DISCUSSION TITLE
How to change the base color of a specific material

### DISCUSSION THREAD

**Discussion by Robeater** [google/model-viewer#2266] [Q&A] [answered]
So I was looking to change the color of a material on my model and I came across [Change Material Base Color](https://modelviewer.dev/examples/scenegraph/#changeColor) example. It seems to work by simply looking through the source file until it comes across material -> pbrMetallicRoughness -> baseColorFactor and changes those values to the ones stored in data-color.

So for my model, it changes the base color of 'BLACK STEEL' when I need to change the color of 'EXTERIOR'. Is there a better way to do this?

![example](https://user-images.githubusercontent.com/48926304/114099444-9f88b980-9877-11eb-8c64-4173851b5059.png)

Thanks in advance.

**Comment by elalish** [comment, marked as answer]
Sorry, our documentation on the scene-graph API is still a bit sparse as it's still under development. This will give you a more complete idea of what's there currently: https://github.com/google/model-viewer/blob/master/packages/model-viewer/src/features/scene-graph/api.ts

There is a list of materials (basically the same list as in the glTF) and each one has a `.name` attribute you can read to check that you have the right one. That example just implicitly grabbed `materials[0]` since there was only one.

**Reply by Robeater** [reply]
Hi Elalish,

That makes a lot more sense now, thanks for the link. 

I know this isn't really the place to ask for JS help, but do you know how I might do that? Currently my if statement just prevents it from executing.

![Code Snippet](https://user-images.githubusercontent.com/48926304/114761389-3f7d9180-9d15-11eb-8b6f-9460d76713ca.png)

For the record it's for a configurator I'm building. All of the models in the carousel have an 'EXTERIOR' material, but at different places in the material list. 

Thanks!

**Reply by elalish** [reply]
you want something like `for (material of modelViewerColor.model.materials) {...}`
